### PR TITLE
[Dataset | Model card] When pushing to template repos, work on actual raw contents

### PR DESCRIPTION
**Pull request by julien-c**
## Question: do we actually want this feature or not?

[Internal Slack convo](https://huggingface.slack.com/archives/C025H22A3ND/p1672064308024229?thread_ts=1670950635.761979&cid=C025H22A3ND)
cc @Wauplin 

### Details

Generated commit for model cards (would need to do the same for datasets): https://huggingface.co/templates/model-card-example/commit/901deccf5acead553f9b082aca480d966e61f355

### CODE REVIEW

**Comment by Wauplin** [comment]
@julien-c  I get the idea of the PR. Not sure what is the optimal way to add a comment to the modelcard. I'm _fine_ with this solution if we don't have any better simple solution. My only concern is that if we have modelcards with a comment in the metadata section, the `modelcards` module is still unable to process them. So if a user uses the `ModelCard` feature of `hfh` at some point, it will remove the comment (is it really an issue though ?).
In any case, if we go with this solution, I would add a comment in `push_repocard_example` to explain shortly why we are not using the module.

**Comment by Wauplin** [comment]
About the failing tests, I don't know what the issue is. I encountered it as well on Friday 23rd afternoon but weren't sure if it was my fault or not. It seems that on staging we cannot commit a `README.md` file from the HTTP endpoint. It gets a `HTTP 400 Bad request for commit endpoint: fetch failed` from moon-landing. It's not about the content of the file but really the filename (I tried it manually and any other filename can be uploaded without any problem.). Maybe something to restart on staging ? cc @julien-c @Pierrci

**Comment by julien-c** [comment]
maybe ping @SBrandeis about the staging env failure (no rush though)

**Comment by julien-c** [comment]
Sleeping on this PR a bit, I think it's an okay approach. No big deal if the actual `ModelCard` strips comments in metadata (which is expected, I guess – YAML => objects conversion doesn't have a concept of comments)

BTW we need to do it for dataset cards too.

Pinging @nateraw for his opinion though, just to be sure

File: utils/push_repocard_examples.py [hunk 1]
```diff
@@ -51,14 +55,22 @@ def push_model_card_example(overwrite: bool) -> None:
     Card is pushed to https://huggingface.co/templates/model-card-example.
     """
 
-    card = ModelCard.from_template(ModelCardData())
+    template = jinja2.Template(ModelCard.default_template_path.read_text())
+    content = template.render(card_data="{}")
```
Comment (nateraw) [inline utils/push_repocard_examples.py:59]: maybe pass description var and include link to raw template along with perhaps some note that that's where the rendered template came from? https://raw.githubusercontent.com/huggingface/huggingface_hub/main/src/huggingface_hub/templates/modelcard_template.md

**Review by nateraw — COMMENTED** [review]
@julien-c this looks great to me! One small issue I noticed here is that we are pushing the rendered template instead of the template itself, so there's not any clear indication in the [pushed model card repo](https://huggingface.co/templates/model-card-example) as to where it came from/how it was created.

Maybe we add a clear link somewhere? Either as a yaml comment or perhaps some header/footer within the card? Perhaps we just pass the `model_summary` jinja template variable when rendering

File: utils/push_repocard_examples.py (hunk 1 above)
Comment (julien-c) [reply utils/push_repocard_examples.py:59]: WDYT @Wauplin ?

Comment (Wauplin) [reply utils/push_repocard_examples.py:59]: I like the idea !

Comment (julien-c) [reply utils/push_repocard_examples.py:59]: do you want to push directly to this branch @nateraw?

Comment (nateraw) [reply utils/push_repocard_examples.py:58]: sure! I see you added a couple commits 11 mins ago - are you done? If so I can push to this branch directly

Comment (julien-c) [reply utils/push_repocard_examples.py:58]: yep i'm done!

**Comment by Wauplin** [comment]
@nateraw @julien-c I have added both `model_summary` and `dataset_summary` to explain from where the templates come from. Could you quickly review and we merge it ? :)

**Review by julien-c — COMMENTED** [review]
can't approve my PR but looks good to me! thx

**Review by Wauplin — APPROVED** [review]

**Pull request merged**